src/gui/tlserrordialog.cpp

```diff
@@ -40,6 +40,12 @@ TlsErrorDialog::TlsErrorDialog(const QList<QSslError> &sslErrors, const QString
 
     connect(_ui->buttonBox, &QDialogButtonBox::accepted, this, &TlsErrorDialog::accept);
     connect(_ui->buttonBox, &QDialogButtonBox::rejected, this, &TlsErrorDialog::reject);
+
+    // Try to set the No-button (the reject button) as the default
+    if (QPushButton *noButton = _ui->buttonBox->button(QDialogButtonBox::No)) {
+        Q_ASSERT(_ui->buttonBox->buttonRole(noButton) == QDialogButtonBox::RejectRole || _ui->buttonBox->buttonRole(noButton) == QDialogButtonBox::NoRole);
```

Comment:
Hmm according to https://doc.qt.io/qt-6/qdialogbuttonbox.html#ButtonRole-enum it is supposed to always be `NoRole`?

Reply:
But how can clicking the "No" button then reject the dialog?

Reply:
I removed the assert, it's superfluous to check the `NoRole` on the "No" button.